Source/Project64/UserInterface/DiscordRPC.cpp

```diff
@@ -0,0 +1,332 @@
+/****************************************************************************
+*                                                                           *
+* Project64 - A Nintendo 64 emulator.                                      *
+* http://www.pj64-emu.com/                                                  *
+* Copyright (C) 2012 Project64. All rights reserved.                        *
+*                                                                           *
+* License:                                                                  *
+* GNU/GPLv2 http://www.gnu.org/licenses/gpl-2.0.html                        *
+*                                                                           *
+****************************************************************************/
+
+#include "stdafx.h"
+#include <3rdParty\discord-rpc\include\discord_rpc.h>
+#include "DiscordRPC.h"
+
+#define UNIX_TIME_START    0x019DB1DED53E8000
+#define TICKS_PER_SECOND   10000000
+
+//Discord Project64 App ID
+#define PJ64_DISCORD_APPID "359086208862650378"
+
+void CDiscord::Init()
+{
+	DiscordEventHandlers handlers = {};
+
+	Discord_Initialize(PJ64_DISCORD_APPID, &handlers, 1, NULL);
+}
+
+void CDiscord::Shutdown()
+{
+	Discord_Shutdown();
+}
+
+void CDiscord::Update(bool bHaveGame)
+{
+	DiscordRichPresence discordPresence = {};
+
+
+	if (strstr(g_Settings->LoadStringVal(Game_GameName).c_str(), "MarioParty3") != NULL)
+	{
+		char szState[256];
+		sprintf(szState, "Playing Mario Party 3");
+		discordPresence.state = szState;
+		discordPresence.startTimestamp = Timestamp();
+		discordPresence.largeImageKey = "mp3";
+		discordPresence.largeImageText = "Mario Party 3";
+		discordPresence.smallImageKey = "icon";
+		discordPresence.smallImageText = "Project64";
+		discordPresence.instance = 1;
+	}
+
+	else if (strstr(g_Settings->LoadStringVal(Game_GameName).c_str(), "MarioParty2") != NULL)
+	{
+		char szState[256];
+		sprintf(szState, "Playing Mario Party 2");
+		discordPresence.state = szState;
+		discordPresence.startTimestamp = Timestamp();
+		discordPresence.largeImageKey = "mp2";
+		discordPresence.largeImageText = "Mario Party 2";
+		discordPresence.smallImageKey = "icon";
+		discordPresence.smallImageText = "Project64";
+		discordPresence.instance = 1;
+	}
```

Comment:
To cut down on repeated code, I suggest adding the `largeImageKey` strings to the RDB file and use Rdb_GoodName for `largeImageText` and `state`.

Reply:
Funny enough I have the RDB good name like that originally for the text and the state: how do I add the strings to the RDB file. Assuming somewhat with. https://github.com/project64/project64/blob/4e2c5949dd6a38eb76c56a09807e8925b604b97b/Source/Project64-core/Settings/SettingsID Also since the key is just the file extension in discord, should I add them to discord via just the RDB name like fully writing out Super Mario 64, or what do you prefer